sync/upstream.py

```diff
@@ -721,13 +721,22 @@ def status_changed(git_gecko, git_wpt, sync, context, status, url, sha):
     if status == "pending":
         # Never change anything for pending
         return landed
-
-    if status == "success":
+    previous_status = sync.last_pr_check
+    success, non_success = env.gh_wpt.status_checks_pass(
+        sync.pr,
+        exclude=["upstream/gecko"]
+    )
+    if success:
+        sync.last_pr_check = "success"
+    else:
+        states = {item.state for item in non_success}
+        sync.last_pr_check = "failure" if "failure" in states else "pending"
+    if success and previous_status != "success":
         sync.error = None
         if sync.gecko_landed():
             landed = sync.try_land_pr()
         else:
-            env.bz.comment(sync.bug, "Upstream web-platform-tests status checked passed, "
+            env.bz.comment(sync.bug, "Upstream web-platform-tests status checks passed, "
```

Comment:
Ideally we only add this comment if we didn't already. Maybe not worth worrying about in this PR.

Reply:
I think with this new approach, repeated comments can still happen but will be much less likely than before in practice.